src/main/groovy/net/neoforged/gradleutils/spotless/ExtractSpotlessConfiguration.groovy

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.gradleutils.spotless
+
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.work.DisableCachingByDefault
+
+import java.nio.file.Files
+
+@CompileStatic
+@DisableCachingByDefault(because = 'configuration contained inside the jar may change')
+abstract class ExtractSpotlessConfiguration extends DefaultTask {
+    @OutputFile
+    abstract RegularFileProperty getOutput()
+
+    ExtractSpotlessConfiguration() {
+        // configuration contained inside the jar may change
+        outputs.upToDateWhen {
+            false
+        }
+    }
+
+    @TaskAction
+    void run() {
+        final output = this.getOutput().get().asFile.toPath()
+        if (Files.notExists(output)) {
+            Files.createDirectories(output.parent)
+        }
+
+        try (
+            final input = SpotlessUtilsExtension.getResourceAsStream('/formatter-config.xml');
+            final out = Files.newOutputStream(output)
+        ) {
+            byte[] buffer = new byte[1024]
+            int len
+            while ((len = input.read(buffer)) > 0) {
+                out.write(buffer, 0, len)
+            }
+        }
```

Comment:
There's _gotta_ be a better way to write this... right? Perhaps some utility which we can make use of? If there's none, I'll be sad, but I'll accept it.